specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/vulnerabilityAssessments.json

```diff
@@ -0,0 +1,199 @@
+                        "name": "resourceNamespace",
+                        "in": "path",
+                        "description": "The Namespace of the resource.",
+                        "required": true,
+                        "type": "string"
+                    },
+                    {
+                        "name": "resourceType",
+                        "in": "path",
+                        "description": "The type of the resource.",
+                        "required": true,
+                        "type": "string"
+                    },
+                    {
+                        "name": "resourceName",
+                        "in": "path",
+                        "description": "Name of the resource.",
+                        "required": true,
+                        "type": "string"
+                    },
+                    {
+                        "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+                    }
+                ],
+                "responses": {
+					"200": {
+                        "description": "OK",
+                        "schema": {
+                            "$ref": "#/definitions/VulnerabilityAssessmentsList"
+                        }
+                    },
+                    "default": {
+                        "description": "Error response describing why the operation failed.",
+                        "schema": {
+                            "$ref": "../../../common/v1/types.json#/definitions/CloudError"
+                        }
+                    }
+                }
+            },
+			"put": {
+				"description": "Creating a vulnerability assessment on a resource, which will onboard a resource for having a vulnerability assessment on it",
+                "tags": [
+                    "VulnerabilityAssessment"
+                ],
+                "operationId": "VulnerabilityAssessment_CreateOrUpdate",
+                "parameters": [
+                    {
+                        "$ref": "../../../common/v1/types.json#/parameters/ResourceId"
+                    },
+                    {
+                        "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "OK",
+                        "schema": {
+                            "$ref": "#/definitions/VulnerabilityAssessment"
+                        }
+                    },
+					"202": {
+                        "description": "Accepted - Resource is being created"
+                    },
+					"default": {
+                        "description": "Error response describing why the operation failed.",
+                        "schema": {
+                            "$ref": "../../../common/v1/types.json#/definitions/CloudError"
+                        }
+                    }
+                }
+            },
+			"delete": {
```

Comment:
DELETE method requires a resource ID, and this path is a collection URI. See comment above about singleton representation.